### PR TITLE
Feature/places special feature api endpoints

### PR DESCRIPTION
**Pull request by fadykstas**
**first merge this pr** => Fix/place feature seed factory #35

Feature/places special feature api endpoints:
- [x] route
- [x] controller
- [x] json httpRequest
- [x] actions
- [x] requests
- [x] responses
- [x] feature test
- [x] route test

### CODE REVIEW

File: app/Actions/Place/SpecialFeature/CollectionSpecialFeatureAction.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+     * @param PlaceFeatureRepositoryInterface $repository
+     */
+    public function __construct(PlaceFeatureRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function execute(CollectionSpecialFeatureRequest $request): CollectionSpecialFeatureResponse
+    {
+        /** @var PlaceFeature $placeFeature */
```
Comment (lenchvolodymyr) [inline app/Actions/Place/SpecialFeature/CollectionSpecialFeatureAction.php]: remove all phpDoc, the code has to be self-documented

File: app/Actions/Place/SpecialFeature/CreateSpecialFeatureAction.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        $placeFeature = $this->repository->save($placeFeature);
+
+        /** @var  CreateSpecialFeatureResponse $response */
+        $response = $this->response = new CreateSpecialFeatureResponse(
+            $placeFeature->id,
+            $placeFeature->name
+        );
+
+        return $response;
+
```
Comment (lenchvolodymyr) [inline app/Actions/Place/SpecialFeature/CreateSpecialFeatureAction.php]: extra enter, and after `}` also
```
        ...
        return ...;
    }
}
```

File: app/Actions/Place/SpecialFeature/CreateSpecialFeatureResponse.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+{
+    protected $id;
+    protected $name;
+
+    /**
+     * CreateSpecialFeatureResponse constructor.
+     * @param $id
+     * @param $name
+     */
+    public function __construct($id, $name)
```
Comment (lenchvolodymyr) [inline app/Actions/Place/SpecialFeature/CreateSpecialFeatureResponse.php]: add type hint for parameters and return values

File: app/Actions/Place/SpecialFeature/DeleteSpecialFeatureAction.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public function execute(DeleteSpecialFeatureRequest $request): DeleteSpecialFeatureResponse
+    {
+        /** @var DeleteSpecialFeatureRequest $request */
+        $id = $request->getPlaceSpecialFeatureId();
+
+        $item = $this->repository->getById($id);
+        if ($item) {
+            $this->repository->deleteById($id);
+        } else {
+            throw new \Exception("Item #$id not found");
```
Comment (lenchvolodymyr) [inline app/Actions/Place/SpecialFeature/DeleteSpecialFeatureAction.php]: Create a special exception, e.g. `SpecialFeatureNotFoundException extends LogicException`

File: app/Http/Controllers/Api/Places/SpecialFeaturesController.php [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+            $responseArray[$id] = [
+                'id' => $id,
+                'name' => $item->getName()
+            ];
+        }
+
+        // base api controller
+        return $this->successResponse($responseArray, 201);
+    }
+
```
Comment (lenchvolodymyr) [inline app/Http/Controllers/Api/Places/SpecialFeaturesController.php]: one enter between methods

File: app/Http/Controllers/Api/Places/SpecialFeaturesController.php [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        CreateSpecialFeatureAction $createSpecialFeatureAction,
+        DeleteSpecialFeatureAction $deleteSpecialFeatureAction,
+        ReadSpecialFeatureAction $readSpecialFeatureAction,
+        CollectionSpecialFeatureAction $collectionSpecialFeatureAction
+    )
+    {
+        $this->createSpecialFeatureAction = $createSpecialFeatureAction;
+        $this->deleteSpecialFeatureAction = $deleteSpecialFeatureAction;
+        $this->readSpecialFeatureAction = $readSpecialFeatureAction;
+        $this->collectionSpecialFeatureAction = $collectionSpecialFeatureAction;
```
Comment (lenchvolodymyr) [inline app/Http/Controllers/Api/Places/SpecialFeaturesController.php]: Incorrect name, name of action consists of `<Verb><Assignment>Action` (e.g. GetCollectionOfSpecialFeaturesAction)

File: app/Http/Controllers/Api/Places/SpecialFeaturesController.php [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+            /** @var CollectionSpecialFeatureResponse $response */
+            $response = $this->collectionSpecialFeatureAction->execute(
+                new CollectionSpecialFeatureRequest()
+            );
+        } catch (\Exception $ex) {
+            // base api controller
+            return $this->errorResponse($ex->getMessage(), 400);
+        }
+
+        $responseArray = [];
```
Comment (lenchvolodymyr) [inline app/Http/Controllers/Api/Places/SpecialFeaturesController.php]: Move this logic to `CollectionSpecialFeaturePresenter`, or better you can implement [JsonSerializable](http://php.net/manual/ru/class.jsonserializable.php) on response

File: app/Http/Controllers/Api/Places/SpecialFeaturesController.php [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    }
+
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $httpRequest
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function store(CreateSpecialFeatureHttpRequest $httpRequest)
```
Comment (lenchvolodymyr) [inline app/Http/Controllers/Api/Places/SpecialFeaturesController.php]: names of methods of controller must be in the imperative notation (e.g. storeSpecialFeature,or createSpecialFeature)

File: app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+use Hedonist\Repositories\Place\PlaceFeatureRepositoryInterface;
+use Hedonist\Entities\Place\PlaceFeature;
+
+
+class CreatePlaceFeatureAction
+{
+    protected $repository;
+
+    /**
```
Comment (lenchvolodymyr) [inline app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php]: Remove all php docs it is not necessary. It has a sense if you use a variable implicitly

File: app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+    public function execute(CreatePlaceFeatureRequest $request): CreatePlaceFeatureResponse
+    {
+        $name = $request->getPlaceFeatureName();
+        $placeFeature = new PlaceFeature([
+            'name' => $name
+        ]);
+        $placeFeature = $this->repository->save($placeFeature);
+
+        return new CreatePlaceFeatureResponse(
```
Comment (lenchvolodymyr) [inline app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php]: Please, indent code, like this:

```
return new CreatePlaceFeatureResponse(
    $placeFeature->id,
    $placeFeature->name
);
```

File: app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+                        $placeFeature->id,
+                        $placeFeature->name
+                    );
+    }
+
```
Comment (lenchvolodymyr) [inline app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php]: remove extra enters

File: app/Http/Controllers/Api/Places/PlaceFeaturesController.php [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function indexPlaceFeature() : JsonResponse
+    {
+        try {
+            $response = $this->collectionPlaceFeatureAction->execute(
+                new GetCollectionPlaceFeatureRequest()
+            );
+        } catch (\Exception $ex) {
+            // base api controller
```
Comment (lenchvolodymyr) [inline app/Http/Controllers/Api/Places/PlaceFeaturesController.php]: such comments delete as well

File: app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php (hunk 9 above)
Comment (fadykstas) [reply app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php]: done

File: app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php (hunk 10 above)
Comment (fadykstas) [reply app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php]: done

File: app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php (hunk 11 above)
Comment (fadykstas) [reply app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php]: done

File: app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php (hunk 10 above)
Comment (fadykstas) [reply app/Actions/Place/SpecialFeature/CreatePlaceFeatureAction.php]: done

**Pull request merged**